test/lib/exec-promise.spec.js

```diff
@@ -0,0 +1,29 @@
+const { expect, assert } = require('chai');
+const exec = require('./../../src/lib/exec-promise');
+const { level } = require('./../../src/lib/log');
+
+describe('exec-promise', () => {
+
+  it('execute command resolve in a promise with the standard output as a result', async () => {
+    const output = await exec(level, 'node', '-h');  // No error is raised
+    expect(output).to.match(/Usage: node/);
+  });
+
+  it('execute command with invalid args raise a rejected promise with output error as a result', async () => {
+    try {
+      await exec(level, 'node', '--invalid-arg');
+      assert.fail('Expected execution error');
+    } catch (err) {
+      expect(err).to.match(/node: bad option/);
+    }
+  });
+
+  it('execute command that does not exist raise a rejected promise', async () => {
+    try {
+      await exec(level, 'cmd-dont-exist.sh', 'some-arg');
+      assert.fail('Expected execution error');
+    } catch (err) {
+      expect(err).to.match(/cmd-dont-exist\.sh: not found/);
+    }
+  });
+});
```

Comment:
Explore mocking out `child_process` so we don't actually exec these commands.

Firstly the ideal unit test checks only the unit you're testing, rather than the whole system. This allows them to be as fast and stable as possible.

Secondly I'm not sure I want tests executing arbitrary code on my machine...

Reply:
Mocked